indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -76,10 +78,21 @@
 
   @Inject
   public TaskLockbox(
-      TaskStorage taskStorage
+      TaskStorage taskStorage,
+      ServerConfig serverConfig
+  )
+  {
+    this.taskStorage = taskStorage;
+    this.lockTimeoutMillis = serverConfig.getMaxIdleTime().getMillis();
+  }
+
+  public TaskLockbox(
+      TaskStorage taskStorage,
+      long lockTimeoutMillis
```

Comment:
`Should every lockAcquireAction have the same timeout?` Yes for now, it can also be extended where client can send a timeout.
For now `serverConfig.getMaxIdleTime() ` is the default timeout b/c even if task gets a lock after this period, overlord can not write the response in the closed socket.